simple-dns/src/dns/rdata/eui.rs

```diff
@@ -0,0 +1,151 @@
+use crate::dns::WireFormat;
+use std::convert::TryInto;
+
+use super::RR;
+
+/// A 48 bit mac address
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+pub struct EUI48 {
+    /// A 48 bit mac address
+    pub address: [u8; 6],
+}
+
+/// A 64 bit mac address
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+pub struct EUI64 {
+    /// A 64 bit mac address
+    pub address: [u8; 8],
+}
+
+impl RR for EUI48 {
+    const TYPE_CODE: u16 = 108;
+}
+
+impl RR for EUI64 {
+    const TYPE_CODE: u16 = 109;
+}
+
+impl<'a> WireFormat<'a> for EUI48 {
+    fn parse(data: &'a [u8], position: &mut usize) -> crate::Result<Self>
+    where
+        Self: Sized,
+    {
+        let address =data[*position..*position + 6].try_into()?;
```

Comment:
Should this care about the byte order?
> The RDATA for an EUI48 RR consists of a single, 6-octet Address field, encoded in network (big-endian) order.

Reply:
As it is decoding it as raw bytes, byte order does not matter.